tetris.py

```diff
@@ -117,59 +116,66 @@ def new_block(self):
         """
         x = int(game_width / 2)
         y = 0
-        blocks = {
-            "L": [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)],
-            "L_rev": [(x - 1, y + 1), (x - 1, y), (x, y), (x + 1, y)],
-            "O": [(x - 1, y), (x, y), (x - 1, y + 1), (x, y + 1)],
-            "E": [(x, y), (x - 1, y + 1), (x, y + 1), (x + 1, y + 1)],
-            "Z": [(x - 1, y), (x, y), (x, y + 1), (x - 1, y + 1)],
-            "Z_rev": [(x + 1, y), (x, y), (x, y + 1), (x - 1, y + 1)],
-            "I": [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)],
-        }
-        if self.upcoming_block is None:
-            self.current_block = random.choice(list(blocks.values()))
+        if self.upcoming_block_shape is None:
+            self.current_block_shape = random.choice(shapes_name)
+            self.current_block_center = (x, y)
         else:
-            self.current_block = self.upcoming_block
-        self.upcoming_block = random.choice(list(blocks.values()))
+            self.current_block_shape = self.upcoming_block_shape
+            self.current_block_center = (x, y)
+        self.upcoming_block_shape = random.choice(shapes_name)
+
+    def get_current_block(self):
+        (x, y) = self.current_block_center
+        if self.current_block_shape == 'I':
+            return [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]
+        if self.current_block_shape == 'L':
+            return [(x - 1, y - 1), (x - 1, y), (x, y), (x + 1, y)]
+        if self.current_block_shape == 'L_rev':
+            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+        if self.current_block_shape == 'O':
+            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+        if self.current_block_shape == 'E':
+            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+        if self.current_block_shape == 'Z':
+            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+        if self.current_block_shape == 'Z_rev':
+            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
 
     def user_input_left(self):
         """
         Moves the current block to the left on the canvas
         """
-        left = []
-        for (x, y) in self.current_block:
-            if x == 0:
-                return
-            left.append((x - 1, y))
-        self.current_block = left
+        x, y = self.current_block_center
+        if x == 0:
```

Comment:
This doesn't work correctly. Instead of checking whether the center is at the very left edge of the game, it should check whether the x coordinate of any square is at the left edge of the game:

```suggestion
        if any(x == 0 for x, y in self.get_current_block()):
```